### PR TITLE
CC-4850: in page nav for docs

### PR DESCRIPTION
**Pull request by WenInCode**
### :hammer_and_wrench: Description

- Adds an in-page-nav component to add a sidebar for navigating a docs page.

**Note:**
- Originally I was trying to get this to auto-generate based on the content, using yielded block and reading in the blocks to generate it, but it was a bit complicated and I realized it's overkill for now.
- Right now this doesn't handle positioning on it's own and it doesn't stick to the top of the page when you scroll. I'm going to be following up with update to the grid system on the page to handle those.

### :camera_flash: Screenshots

**When it's below the width media query it will set remove itself**
Screenshot 2023-05-12 at 3 03 37 PM

**Here it be**
Screenshot 2023-05-12 at 4 53 32 PM

---

**Example of the link working**

### :link: External Links

[CC-4850](https://hashicorp.atlassian.net/browse/CC-4850)

### :+1: Definition of Done

- [x] New functionality works?
- [x] Tests added?
- [x] Docs updated?

### :speech_balloon: Using the [Netlify feedback ladder](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify/)

- :mount_fuji: **[mountain]**: Blocking and requires immediate action.
- :moyai: **[boulder]**: Blocking.
- :white_circle: **[pebble]**: Non-blocking, but requires future action.
- :hourglass_flowing_sand: **[sand]**: Non-blocking, but requires future consideration.
- :sparkles: **[dust]**: Non-blocking. "Take it or leave it"

[CC-4850]: https://hashicorp.atlassian.net/browse/CC-4850?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by WenInCode** [comment]
I figured out the sticky css

File: documentation/app/router.js [hunk 1]
```diff
@@ -4,7 +4,9 @@
 
 import EmberRouter from '@ember/routing/router';
 import config from 'documentation/config/environment';
+import { withHashSupport } from 'ember-url-hash-polyfill';
 
+@withHashSupport
```
Comment (natmegs) [inline documentation/app/router.js:9]: TIL!

File: documentation/app/templates/components/list-item.hbs [hunk 2]
```diff
@@ -20,209 +20,235 @@
...
+          list item “state” =“default” in order to have interactivity within
+          prototypes and code.</p>
 
-    <h2>Interactivity & Actions</h2>
-    <h3>Interactive List Item</h3>
-    <p>The entirety of List Item can be actionable (ie. it can be hovered,
-      clicked, and focused), this is the most common behavior. Ensure the list
-      item “state” =“default” in order to have interactivity within prototypes
-      and code.</p>
+        <h3>Static List Item</h3>
```
Comment (natmegs) [inline documentation/app/templates/components/list-item.hbs:57]: Does this header (and others) need an `id` attribute to get the browser to scroll to position when the nav items are clicked? It's working great for me for the "When to use" and "Example section" links 👍

**Review by natmegs — APPROVED** [review]
Looks great!!

File: documentation/app/templates/components/list-item.hbs (hunk 2 above)
Comment (WenInCode) [reply documentation/app/templates/components/list-item.hbs:57]: You are right 🤦

**Pull request merged**